### PR TITLE
Handle malformed payload for LinuxPackageOpened

### PR DESCRIPTION
**Pull request by antoinehashbang**
https://phabricator.endlessm.com/T32200

### CODE REVIEW

File: azafea/event_processors/endless/metrics/tests_v2/test_events.py [hunk 1]
```diff
@@ -650,6 +650,42 @@ def test_hack_clubhouse_progress_event_with_unknown_key():
     assert not hasattr(event, 'unknown')
 
 
+def test_valid_linux_package_opened():
+    from azafea.event_processors.endless.metrics.v2.model import LinuxPackageOpened
+
+    payload = GLib.Variant('mv', GLib.Variant('s', 'gnome-calendar.deb'))
+    lpo = LinuxPackageOpened(payload)
+    assert lpo.argv == ['gnome-calendar.deb']
```
Comment (dbnicholson) [inline azafea/event_processors/endless/metrics/tests_v2/test_events.py:658]: Could you add a second test here that creates the payload in the expected `as` format to ensure it gets through the parsing correctly?

File: azafea/event_processors/endless/metrics/v2/model/__init__.py [hunk 2]
```diff
@@ -775,10 +776,29 @@ class LinuxPackageOpened(SingularEvent):
...
+        payload = maybe_payload.get_maybe()
+
+        if payload is None:
+            raise EmptyPayloadError(f'Metric event {self.__event_uuid__} needs a '
+                                    f'{self.__payload_type__} payload, but got none')
+
+        payload = get_variant(payload)
+        payload_type = payload.get_type_string()
+
+        if payload_type != self.__payload_type__ and payload_type != 's':
```
Comment (dbnicholson) [inline azafea/event_processors/endless/metrics/v2/model/__init__.py:789]: Could you add a comment explaining this?

File: azafea/event_processors/endless/metrics/v2/model/__init__.py [hunk 3]
```diff
@@ -775,10 +776,29 @@ class LinuxPackageOpened(SingularEvent):
...
+            raise WrongPayloadError(f'Metric event {self.__event_uuid__} needs a '
+                                    f'{self.__payload_type__} payload, but got '
+                                    f'{payload} ({payload_type})')
+        return self._get_fields_from_payload(payload)
+
     @staticmethod
     def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        value = payload.unpack()
+        if type(value) == str:
```
Comment (dbnicholson) [inline azafea/event_processors/endless/metrics/v2/model/__init__.py:798]: I don't think it makes any real difference, but I prefer `isinstance(value, str)`.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
I left some minor comments, but this looks great. Thanks!

**Review by dbnicholson — APPROVED** [review]
This looks good to me. I'm going to merge this since it seems correct. @liZe when you get a chance could you take a look at this? I'm not familiar with the models here and want to make sure there's nothing wrong with the method overrides.

**Comment by liZe** [comment]
Everything looks fine for me!

**Pull request merged**